src/redux/local-storage/search-equipment-history.tsx

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+import { UUID } from 'crypto';
+import { mixed, string } from 'yup';
+import { EquipmentInfos, EquipmentType } from '@gridsuite/commons-ui';
+import { APP_NAME } from 'utils/config-params';
+import yup from 'components/utils/yup-config';
+
+const MAX_SEARCH_EQUIPMENT_HISTORY_SIZE = 5;
+
+const LOCAL_STORAGE_SEARCH_EQUIPMENT_HISTORY_KEY = (
+    APP_NAME + '_SEARCH_EQUIPMENT_HISTORY_'
+).toUpperCase();
+
+const getLocalStorageSearchEquipmentHistoryKey = (studyUuid: UUID) => {
+    return LOCAL_STORAGE_SEARCH_EQUIPMENT_HISTORY_KEY + studyUuid;
+};
+
+const equipmentSchema = yup.object().shape({
```

Comment:
Nice for type checking at running !
Maybe we should share with the team about this technique ?

Reply:
I agree, even if this is something we won't often do